File: FrontEnd/xego_driver/lib/models/Dto/login_response_dto.dart
import 'package:xego_driver/models/Dto/tokens_dto.dart';
import 'package:xego_driver/models/Dto/user_dto.dart';

class LoginResponseDto {
  final UserDto userDto;
  final TokensDto tokensDto;
  final bool isSuccess;
  final String message;

  LoginResponseDto({
    required this.userDto,
    required this.tokensDto,
    required this.isSuccess,
    required this.message,
  });

  factory LoginResponseDto.fromJson(Map<String, dynamic> json) {
    return LoginResponseDto(
      userDto: UserDto.fromJson(json['data']['user']),
      tokensDto: TokensDto.fromJson(json['data']['tokens']),
      isSuccess: json['isSuccess'],
      message: json['message'],
    );
  }
}

File: BackEnd/XeGo-Microservices/Services/Auth/XeGo.Services.Auth.API/Models/Dto/LoginResponseDto.cs
namespace XeGo.Services.Auth.API.Models.Dto
{
    public class LoginResponseDto
    {
        public UserDto? User { get; set; }
        public TokenDto Tokens { get; set; } = new();
        public List<String> Roles { get; set; } = new();
    }
}
